tck/src/main/java/org/eclipse/microprofile/lra/tck/LraTckConfigBean.java

```diff
@@ -34,6 +34,45 @@
      */
     private static final Long LRA_TIMEOUT_MILLIS = 50000L;
 
+    /**
+     * Name of the system property that is used as factor adjusting timeout values
+     * used in the testsuite.
+     */
+    private static final String LRA_TCK_TIMEOUT_FACTOR_PROPETY_NAME = "lra.tck.timeout.factor";
+
+    /**
+     * Name of the system property which is used to configure the TCK base url.
+     * See {@link LraTckConfigBean#tckSuiteBaseUrl}.
+     */
+    private static final String LRA_TCK_BASE_URL_PROPERTY_NAME = "lra.tck.base.url";
+
+    /**
+     * A helper method which is capable to adjust timeout value by the provided factor.
+     *
+     * @param timeout  timeout value which will be adjusted
+     * @param timeoutFactor  factor that will be used to adjusting the timeout (by multiplying it)
+     * @return  adjusted timeout by te provided factor
+     */
+    public static long adjustTimeout(long timeout, double timeoutFactor) {
+        if (timeout < 0 || timeoutFactor < 0)  {
+            throw new IllegalArgumentException(String.format(
+                    "Provided arguments (timeout=%d, timeoutFactor=%.2f) has to be positive", timeout, timeoutFactor));
+        }
+        return (long) Math.ceil(timeout * timeoutFactor);
+    }
+
+    /**
+     * A helper method which is capable to adjust timeout value
+     * by factor obtained from system property {@code #LRA_TCK_TIMEOUT_FACTOR_PROPETY_NAME}.
+     *
+     * @param timeout  timeout value which will be adjusted
+     * @return  adjusted timeout by te default factor
+     */
+    public static long adjustTimeoutByDefaultFactor(long timeout) {
+        String timeoutFactor = System.getProperty(LraTckConfigBean.LRA_TCK_TIMEOUT_FACTOR_PROPETY_NAME, "1.0");
```

Comment:
use please the config property from line https://github.com/eclipse/microprofile-lra/pull/313/files#diff-d7c8d58d8c6caa6639e33500b0985073R87 as that can be also taken from env vars, file, etc. not only system properties.

Reply:
@xstefank that's the trouble :-) I absolutely understand what you talk about and I was a bit hesitating to propose this change. The reason why I provided this method which uses the system property lookup is the `TckRecoveryTests` which run out of the container. Because of that there is no MP config available
See here: https://github.com/eclipse/microprofile-lra/pull/313/files#diff-0148f6b901fbaaf1ace873d4d13ce058R189

As I'm thinking about it once again I think it's a bit misleading and not a good idea to add this method under the config bean at the first place. The config bean should be just operating with MP Config stuff.
I can see two ways how to change this this - either completely omit the tck timeout factor from the `TckRecoveryTests` or (and I would probably prefer this option) adding the `System.getProperty` call only for the `TckRecoveryTests` class and not into the `LraTckConfigBean`. Would that be ok by you?

Reply:
@xstefank I did the adjustment with system property used only in the `TckRecoveryTests`. Please, take a look.

Reply:
I wonder whether it must then be marked as run as client? In every case, document please that this one must be a system property in the `running_the_tck.asciidoc` please.